src/web/core/src/main/java/org/geoserver/web/wicket/GeoServerDataProvider.java

```diff
@@ -43,6 +41,8 @@
  * @param <T>
  */
 public abstract class GeoServerDataProvider<T> extends SortableDataProvider<T, Object> {
+    public static final Pattern EXACT_TERM_KEYWORD_PATTERN =
+            Pattern.compile("^([\"'])(?<keyword>.+)\\1$");
```

Comment:
This one needs some explanation, something like:

```
// matches exact terms enclosed by either double or single quotes
// The first matching group matches either " or ', the "keyword" matching group matches any sequence of at least one character, the last group refers to the first one (\\1$) ensuring the start and end quote are balanced
```
A few more notes:
* Does the field need to be public? If not, default to private or protected.
* The documentation does not mention single quote being a possibility, please amend either docs or implementation, or people will wonder why searching for ``un'azione dell'altro mondo`` weirdly only matches ``azione dell`` but not ``azione``. Given the single quote has legit usages, I'd go for removing it from the regex, which will also simplify it
* I don't see a way to escape the double quote character, should it happen to be a valid value in the string. This could also be addressed by documentation, stating explicitly that there is no escaping support.

Reply:
Added a Javadoc for the field and made it private. Thanks for the feedback.

Regarding the search query behavior, if a query such as 'un'azione dell'altro mondo` is entered, the query will be tokenized by splitting on spaces, resulting in the search terms: `["un'azione", "dell'altro", "mondo"]`. This maintains the legitimate use of the single quote within the words.

Additionally, the double quote character could still be part of the string. For example, searching for `un'azione dell'altro "mondo` would produce the keywords: `["un'azione", "dell'altro", ""mondo"]`.